src/storage/cpu_shared_storage_manager.h

```diff
@@ -91,11 +95,18 @@ class CPUSharedStorageManager final : public StorageManager {
  private:
   static constexpr size_t alignment_ = 16;
 
-  std::mutex mutex_;
+  std::recursive_mutex mutex_;
   std::mt19937 rand_gen_;
-  std::unordered_map<void*, Storage::Handle> pool_;
+  std::unordered_map<void*, Storage::Handle> pool_; 
+#ifdef _WIN32
+  std::unordered_map<void*, Storage::Handle> is_free_;
+  std::unordered_map<void*, HANDLE> map_handle_map_;
```

Comment:
why do you need two more maps?

Reply:
only use pool_ has the problem of repeated release.